### PR TITLE
Document how to disambiguate symbol links using type signature information 

### PR DESCRIPTION
**Pull request by d-ronnqvist**
Bug/issue #, if applicable: rdar://136207820 

## Summary

This adds user facing documentation about disambiguating symbol links using type signature information, added in https://github.com/swiftlang/swift-docc/pull/643 and https://github.com/swiftlang/swift-docc/pull/1087

At first I tried to insert a section about type signature into the existing documentation structure but I found that the current order made the documentation hard to follow. Specifically, these two changes made the flow of the documentation somewhat confusing:

- Examples about unambiguous symbol links were both before and after the examples about ambiguous symbol links. With even more examples about ambiguous links, this made it hard to know where to insert the new examples.
- Disambiguation using hashes was explained before disambiguation using symbol types. This made it hard to know where to insert the explanation about type signature disambiguation.

So, before adding new documentation I addressed some of the issues with the flow of the existing documentation:
- Adding more documentation, with examples, about how relative symbol links work.
- Moving all examples of unambiguous symbol links before any examples of ambiguous symbol links and creating a dedicated subsection about link disambiguation.
- Move the example about symbol kind disambiguation before the example about hash disambiguation so that there's a clear flow from simpler and more readable disambiguation to the unreadable hash disambiguation. 
- Add a handful of missing symbol type to the list of supported symbol type disambiguation values.

With those changes done, it felt more natural to:
- First update the `Sloth/update(_:)` to give a basic description of type signature disambiguation
- Then add a more advanced example with mixed parameter type and return values disambiguation. 

## Dependencies

None.

## Testing

Read the new documentation.

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- ~[ ] Added tests~ This only changes documentation.
- [x] Ran the `./bin/test` script and it succeeded
- [x] Updated documentation if necessary

### CODE REVIEW

**Review by heckj — APPROVED** [review]
Looks great David. Kind of jealous of the text-graphics skill in some of those examples. And **hugely** looking forward to this capability in 6.1+, the symbols are hugely easier to understand while writing!

File: Sources/docc/DocCDocumentation.docc/linking-to-symbols-and-other-content.md [hunk 1]
```diff
@@ -103,65 +150,146 @@ The following example shows a symbol link to the `red` static property:
...
-Symbol type suffixes can include a source language identifier prefix — for 
-example,  `-swift.enum` instead of `-enum`. However, the language 
-identifier doesn't disambiguate the link.
+You can discover these symbol type suffixes from DocC's warnings about ambiguous symbol links.
+DocC suggests one disambiguation for each of the symbols that match the ambiguous symbol path.
 
-Symbol paths are case-sensitive, meaning that symbols with the same name in
-different text casing don't need disambiguation. 
+Symbol type suffixes can include a source language identifier prefix---for example, `-swift.enum` instead of `-enum`. 
+However, the language identifier doesn't disambiguate the link.
```
Comment (mayaepps) [inline Sources/docc/DocCDocumentation.docc/linking-to-symbols-and-other-content.md:190]: If it doesn't disambiguate the link, why would someone use this suffix?

**Review by mayaepps — APPROVED** [review]
🎉

File: Sources/docc/DocCDocumentation.docc/linking-to-symbols-and-other-content.md (hunk 1 above)
Comment (d-ronnqvist) [reply Sources/docc/DocCDocumentation.docc/linking-to-symbols-and-other-content.md:190]: In the first few releases of DocC (before Xcode 14.3) DocC required the language identifier prefix for unrelated and uninteresting implementation detail reasons, even though the that information wasn't useful for disambiguation purposes. Links created during those releases continue to work but can be shortened.
